src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -249,15 +220,19 @@ function processInlineTag(
       .toOrderedSet();
   }
   return currentStyle;
-}
+};
 
-function joinChunks(A: Chunk, B: Chunk): Chunk {
+const joinChunks = (
+  A: Chunk,
+  B: Chunk,
+  experimentalHasNestedBlocks?: boolean,
+): Chunk => {
   // Sometimes two blocks will touch in the DOM and we need to strip the
   // extra delimiter to preserve niceness.
-  var lastInA = A.text.slice(-1);
-  var firstInB = B.text.slice(0, 1);
+  const lastInA = A.text.slice(-1);
+  const firstInB = B.text.slice(0, 1);
 
-  if (lastInA === '\r' && firstInB === '\r') {
+  if (lastInA === '\r' && firstInB === '\r' && !experimentalHasNestedBlocks) {
```

Comment:
This is the area where we updated in the `joinChunks` section. This is responsible for previously flattning blocks making nested blocks content collapse to root container elemenets. Since we want to support nested blocks we don't want this to happen when the flag is enabled